presto-main/src/main/java/com/facebook/presto/metadata/GlobalProperties.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.metadata;
+
+import com.facebook.presto.memory.NodeMemoryConfig;
+import io.airlift.units.DataSize;
+
+import static java.util.Objects.requireNonNull;
+
+public final class GlobalProperties
+{
+    private final DataSize maxMemoryPerNode;
+
+    public GlobalProperties()
+    {
+        this(new NodeMemoryConfig());
+    }
+    public GlobalProperties(NodeMemoryConfig nodeMemoryConfig)
```

Comment:
I thought there might be other relevant properties (anyway that would need to be passed through the same way global properties is here... the point is that NodeMemoryConfig wasn't accessible to the optimizer).  Also, I wanted it to be clear that if something becomes a session property, that this is no longer the right way to call it.